### DISCUSSION TITLE
Question on neighbour pairs calculation and interpretation.

### DISCUSSION THREAD

**Discussion by SunilAnandatheertha** [mtex-toolbox/mtex#2218] [Ask Anything]
Hello all, I seek clarification in the way neighbours are calculated. I am trying to assimilate the list of all grain IDs which fall as neighbours for a given central grain. I give the following piece of code that I am using. 

```matlab
[grains, ebsd.grainId] = calcGrains(EBSDMAP('indexed'), 'angle', 5*degree);
pairs = grains.neighbors;
grainID = 201;
neighbors = getNeighbors(grainID, pairs);
plot(grains, grains.meanOrientation)
hold on
plot(grains(neighbors).boundary,  'LineWidth', 4, 'linecolor', 'b', 'LineStyle', ':')
plot(grains(grainID).boundary, 'LineWidth', 4, 'linecolor', 'k')
hold off
```
The function getNeighbours is provided below. It only assimilates all neighbouring grain IDs from the pairs variable for a given grainID of interest.:
```matlab
function neighbors = getNeighbors(grainID, pairs)
    idx = pairs(:,1) == grainID;
    neighbors = pairs(idx, 2);   
    neighbors = unique(neighbors);
end
```
![image](https://github.com/user-attachments/assets/1e5d6875-cab3-4db5-99d6-4e138ce50606)

1. Can someone please explain the criteria used in neighbour condition.
2. Can someone please help me understand the if a grain touches another at a single point, would they be considered as neighbours or not. 
3. Alo in the image above, for the grain of interest, I would expect grain 1 to be one among the identified neighbours no?
Thank you 
Sunil

**Comment by ralfHielscher** [comment]
two grains are considered as neighbors if they share a common grain boundary. If they meet on in a single points they are not considered as neighbors. It is strange that in your example grain 1 is not considered as neighbor. Maybe you can provide the data?

Ralf.

**Reply by SunilAnandatheertha** [reply]
Hi Ralf, thank you for the reply. Unfortunately, I cannot share that data. I will try and find something and reply back.
